### PR TITLE
Update metadata after transferring a job

### PR DESCRIPTION
**Pull request by fejta**
ref https://github.com/kubernetes/test-infra/issues/11322 https://github.com/kubernetes/test-infra/issues/11009

gubernator: https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-test-infra-bazel/76414/
spyglass: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-test-infra-bazel/76414

/assign @Katharine @krzyzacy

### CODE REVIEW

**Comment by Katharine** [comment]
It looks like Spyglass can't look up the GCS path for that prowjob for some reason. The GCS version works okay: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-test-infra-bazel/76414

Hm.

**Comment by Katharine** [comment]
"Some reason" looks to be the combination of "it was undecorated" and "the pod has gone away".

**Comment by fejta** [comment]
/pony pinky pie

Can has reveew plz?

File: experiment/resultstore/main.go [hunk 1]
```diff
@@ -228,18 +231,107 @@ func transferLatest(ctx context.Context, storageClient *storage.Client, rsClient
...
+	}
+	if err := updateStarted(ctx, storageClient, path, result.started); err != nil {
+		return fmt.Errorf("update started.json: %v", err)
+	}
+	return nil
+}
+
+const (
+	linksKey       = "links"
+	resultstoreKey = "resultstore"
```
Comment (Katharine) [inline experiment/resultstore/main.go]: Should this be "ResultStore"?

Comment (fejta) [reply experiment/resultstore/main.go]: lowercase forever :metal:

**Comment by fejta** [comment]
Kath, can you re-lgtm? Merge conflict with your gcs public read PR (notw fixed)

**Pull request merged**